Sources/GRPC/Interceptor/ClientInterceptorPipeline.swift

```diff
@@ -270,13 +270,19 @@ private extension InterceptorContextList {
     onRequestPart: @escaping (GRPCClientRequestPart<Request>, EventLoopPromise<Void>?) -> Void,
     onResponsePart: @escaping (GRPCClientResponsePart<Response>) -> Void
   ) where Element == ClientInterceptorContext<Request, Response> {
-    let middle = interceptors.enumerated().map { index, interceptor in
-      ClientInterceptorContext(
-        for: .userProvided(interceptor),
-        atIndex: index,
-        in: pipeline
-      )
-    }
+    let middle = { () -> [Element] in
+      if interceptors.count == 0 {
+        return [] // Special case 0 to reduce cost.
+      } else {
+        return interceptors.enumerated().map { index, interceptor in
+          ClientInterceptorContext(
+            for: .userProvided(interceptor),
+            atIndex: index,
+            in: pipeline
+          )
+        }
+      }
+    }()
```

Comment:
Why is this a self-invoking block?

Reply:
Because I keep forgetting how smart swift is - changed it.